src/azul/service/async_manifest_service.py

```diff
@@ -63,41 +75,88 @@ class AsyncManifestService(AbstractService):
+                         token: Optional['Token'] = None,
+                         execution_id: Optional[str] = None,
+                         retry_index: Optional[int] = 0) -> Token:
+        if execution_id is None:
+            _, _, execution_id = input["object_key"].rpartition('/')
+            execution_id, _, _ = execution_id.partition('.')
+            assert '.' not in execution_id, execution_id
+        if token is not None:
+            execution_id = token.execution_id
+            execution_id, _, _ = execution_id.partition('_')
+            assert '_' not in execution_id, execution_id
+        execution_id += f'_{retry_index}'
+        if retry_index > self.step_function_execution_retries:
+            raise StateMachineError('Maximum manifest generation retries reached.',
+                                    execution_id)
+        token = Token(execution_id=execution_id,
+                      request_index=0,
+                      wait_time=self._get_next_wait_time(0))
         try:
-            execution = self.step_function_helper.describe_execution(state_machine_name=self.state_machine_name,
-                                                                     execution_name=token.execution_id)
+            self.step_function_helper.start_execution(self.state_machine_name,
+                                                      execution_id,
+                                                      execution_input=input)
         except ClientError as e:
-            if e.response['Error']['Code'] == 'ExecutionDoesNotExist':
-                raise InvalidTokenError from e
+            if e.response['Error']['Code'] == 'ExecutionAlreadyExists':
+                token = self.retry_generation(input, token, execution_id, retry_index)
             else:
                 raise
-        output = execution.get('output', None)
-        status = execution['status']
-        if status == 'SUCCEEDED':
+        return token
+
+    def retry_generation(self,
+                         input: JSON,
+                         token: 'Token',
+                         execution_id: str,
+                         retry_index: int):
+        output, status = self._inspect_generation(token)
+        if status is StepFunctionStatus.running:
+            return token.advance(wait_time=self._get_next_wait_time(token.request_index))
+        elif status in (StepFunctionStatus.failed,
+                        StepFunctionStatus.succeeded,
+                        StepFunctionStatus.timed_out,
+                        StepFunctionStatus.aborted):
+            if retry_index < self.step_function_execution_retries:
+                return self.start_generation(input,
+                                             token=token,
+                                             execution_id=execution_id,
+                                             retry_index=retry_index + 1)
+        raise StateMachineError(status, output)
```

Comment:
This was a mistake on my part. Adding `else` changes the semantics here because the `elif` branch only returns if an additional condition is met. I think there's still room for improvement here but it may be above my station to determine what form it should take. For now, I suggest reverting to what you had before.